qiskit/ignis/discriminator/discriminator.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from abc import abstractmethod
+
+from qiskit.result import Result
+
+
+class AbstractDiscriminator(object):
+    """Abstract discriminator class that hold the methods signatures
+    used for discrimination, e.g. to convert IQ data into 0 and 1's."""
+
+    @abstractmethod
+    def __init__(self, result: Result, cal_circuits: list, data_circuits: list,
```

Comment:
For each `cal_circuit` I think you will need the associated state that it is a calibration circuit for. This will be an integer or string of length memoryslots `integer_state = bin(state, 2)`.